common/chat.cpp

```diff
@@ -1769,6 +1890,11 @@ static common_chat_params common_chat_templates_apply_jinja(
         return common_chat_params_init_command_r7b(tmpl, params);
     }
 
+    // Granite (IBM) - detects thinking support
+    if (src.find("elif thinking") != std::string::npos && src.find("<think>") != std::string::npos) {
```

Comment:
```suggestion
    // Granite (IBM) - detects thinking / tools support
    if (src.find("elif thinking") != std::string::npos && src.find("<|tool_call|>") != std::string::npos) {
```
This should be less likely to misidentify the template.

Reply:
Thank you for this. I've updated the code with the suggested change and merged with the latest.